Add Kyma Binding pipelines docs (#9935)
kyma-project/kyma: components/binding/docs/03-binding-pipelines.md

# Kyma Binding Pipelines

> During the development of the Kyma Binding component we decided to change the pipelines from the regular jobs run on Prow. Main focus was to make them faster. The idea was to run the job on k3s, since it's objectively the [best](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use), and install only the Binding component, without the whole Kyma. The decision to use Github Actions was made since the k3s scripts were already [developed](https://www.facebook.com/groups/1345683025593684/permalink/1636584193170231/). Main problem for this approach was that GitHub doesn't propagate the secrets created on the main repository, to pull requests opened from forked repositories. As the workflow in Kyma is based on using forks we had to create a workaround for that. The idea for that was found [here](https://github.com/imjohnbo/ok-to-test). The jobs that we developed:

Reviewer: I'd twist the overview part a bit and focus on the facts - simply state which infrastructure is used for this component. I'd also avoid using 2nd person plural ("we"). Let's discuss this.

> During the development of the Kyma Binding component we decided to change the pipelines from the regular jobs run on Prow. Main focus was to make them faster. The idea was to run the job on k3s, since it's objectively the [best](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use), and install only the Binding component, without the whole Kyma. The decision to use Github Actions was made since the k3s scripts were already [developed](https://www.facebook.com/groups/1345683025593684/permalink/1636584193170231/). Main problem for this approach was that GitHub doesn't propagate the secrets created on the main repository, to pull requests opened from forked repositories. As the workflow in Kyma is based on using forks we had to create a workaround for that. The idea for that was found [here](https://github.com/imjohnbo/ok-to-test). The jobs that we developed:

Author: Of course not, sorry it was kind of late 😅

> ## [Is trusted dev](https://github.com/kyma-project/kyma/blob/master/.github/workflows/trusted-dev.yaml)

Author: I wanted to make it easier to see the workflow file, do you see any other way for that or should I leave it?

> ## [Is trusted dev](https://github.com/kyma-project/kyma/blob/master/.github/workflows/trusted-dev.yaml)

Author: Ok, I didn't see the rest of the review, that looks much better

This job is opened on every PR to the `components/binding` directory. It check if the author of the PR has write access to the Kyma repo. If it has it sends a `repository-dispatch` event which triggers the main pipeline

## [Ok to test](https://github.com/kyma-project/kyma/blob/master/.github/workflows/ok-to-test.yaml)

This job allows to run the pipeline on PRs made by external contributors. A reviewer has to check whether the PR doesn't contain any malicious code and then put a comment with `/ok-to-test` command and then put the commit hash of the HEAD commit of the PR, eg.

```
/ok-to-test sha=44fb1091e18f025a365495c2d268bce944b4239f
```

Such comment will trigger the main pipeline

## [Pre-master binding k3s](https://github.com/kyma-project/kyma/blob/master/.github/workflows/pre-master-binding-k3s.yml) and [Post-master binding k3s](https://github.com/kyma-project/kyma/blob/master/.github/workflows/post-master-binding-k3s.yml)

Those are the main integration jobs for the Kyma Binding component. Their responsibility is to

> - go test, vet, and fmt the code

Reviewer: Where's the verb here? 🙂

# Kyma Binding pipelines

> Kyma Binding pipelines are based on Github Actions and use k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. The script to run k3s is based on [this script](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a workaround based on [this solution](https://github.com/imjohnbo/ok-to-test). For this purpose, the following jobs were developed:

Reviewer: ```suggestion
Kyma Binding pipelines are based on GitHub Actions and use k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. The script to run k3s is based on [this script](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a workaround based on [this solution](https://github.com/imjohnbo/ok-to-test). For this purpose, the following jobs were developed:
```

> Kyma Binding pipelines are based on Github Actions. This decision is due to the fact that scripts to run k3s are already [developed](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh) and k3s is objectively the [best](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run on CI. Main problem for this approach was that GitHub doesn't propagate the secrets created on the main repository, to pull requests opened from forked repositories. As the workflow in Kyma is based on using forks we had to create a workaround for that. The idea for that was found [here](https://github.com/imjohnbo/ok-to-test). For this purpose, we developed the following jobs:

Reviewer: ```suggestion
Kyma Binding pipelines are based on Github Actions and k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. You can see the script to run k3s [here](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a [workaround](https://github.com/imjohnbo/ok-to-test) for that. For this purpose, the following jobs were developed:
```
